pages/u/[username]/index.tsx

```diff
@@ -118,7 +119,15 @@ export const getServerSideProps = async (context: UserSSRPropsContext) => {
   });
 
   if (!req.ok) {
-    return { notFound: true };
+    if (req.status === 404) {
```

Comment:
This logic makes sense in the page @brandonroberts @jpmcb, but the reality is if the API is bogged down, it fails to get the user info for a valid user and will error (good), but if it's an invalid user, we 400 instead of 404. TLDR; we'll never get a 404 if `!req.ok`. The check up further for a valid slug will 404 if it's not a valid github username though.

TLDR; this is a good fix if a valid user profile fails to load, but for a non existent user with a valid GitHub username format will also throw.

Reply:
This sounds like the API needs to handle its errors better instead of the App trying to conform to the API's poor error handling. Taking a quick look at the code there, if the GitHub client code fails during user stubbing (for any reason like a 500, 400, etc.), the API will just throw a bad request exception.

This could probably have a pattern match on the GitHub request status code and bubble that up from there (i.e., if GitHub 404s on a user not being there, we thrown that not found exception).

Reply:
Yeah definitely some changes to the API are in order, but on the page level, I think the logic here makes sense.